src/client/Client.cc

```diff
@@ -8523,9 +8527,11 @@ int Client::readdir_r_cb(dir_result_t *d, add_dirent_cb_t cb, void *p,
     uint64_t next_off = 1;
 
     int r;
+    caps |= CEPH_STAT_RSTAT;
     r = _getattr(diri, caps, dirp->perms);
     if (r < 0)
       return r;
+    caps &= ~CEPH_STAT_RSTAT;
```

Comment:
I'm not sure about cleaing the bits here. It's possible to have CEPH_CAP_RSTAT already set in the caps mask, and this just indiscriminately removes them. It'd probably be better to add a new "mask" variable or locally scoped to this block:

    `int mask = caps | CEPH_STAT_RSTAT;`

...and then you won't be messing with the "caps" value.

Reply:
It's fixed，thanks!